tactics/class_tactics.ml

```diff
@@ -442,6 +442,17 @@ let make_hints env sigma (modes,st) only_classes sign =
       else hints)
     sign db
 
+module Intpart = Unionfind.Make(Evar.Set)(Evar.Map)
+
+type solver =
```

Comment:
Could we get a record wrapper here? A function with that many arguments is prone to generate unreadable error messages if we modify the API.

Reply:
Ok, but that is the type of the built-in solver as well. You want just the public api to be wrapped, or the internal one as well?

Reply:
And to help @FissoreD please give a pointer to a wrapped api, I don't know exactly myself what you want exactly :-)

Reply:
At least the public API.

Reply:
Just something along the lines of
```ocaml
type solver = { solver : 
  Environ.env ->
  Evd.evar_map ->
  depth:int option ->
  unique:bool ->
  best_effort:bool ->
  goals:Evar.t list ->
  (bool * Evd.evar_map)
}
```

Reply:
@ppedrot in my last commit I wrap the type solver in a record :+1: